
Introduction
============

Deconvolution problems are commonplace in many areas of science and engineering. In the context of biomedical research, a typical example is how to isolate specific gene expressions of distinct populations (cell types, tissues, and genes) from composite measures obtained by a single analyte or sensor. This problem is often a consequence of the prohibitive cost of profiling each population separately [@subramanian2017next; @cleary2017efficient] and has important implications for the analysis of transcriptional data in mixed samples [@shen2010cell; @zhong2012gene; @newman2015robust; @zaitsev2019complete], single-cell data [@deng2019scalable], the study of cell dynamics [@lu2003expression], but also for the analysis of imaging data [@preibisch2014efficient]. 

<!-- Deconvolution problems are ubiquitous in many areas of science and engineering. In the context of biomedical research, the issue often consists of isolating specific gene expression levels from distinct populations (cell types, tissues, and genes) using composite measures obtained from a single analyte or sensor; a problem that stems from the prohibitive cost of profiling each population separately [@subramanian2017next; @cleary2017efficient]. Typical examples include measuring cell-type specific differential gene expression from mixed samples [], profiling single-cell gene expression in heterogeneous tissues [@deng2019scalable], and tracking cell growth in dynamically evolving populations []; but similar problems arise also in many other related contexts, such as imaging [e.g., multiview integration, @preibisch2014efficient]. In all these situations, researchers must deconvolute the composite data to identify and isolate measurements from each population.Â  -->

Existing deconvolution approaches work well in many specific settings, but might be suboptimal to the increasing availability of biological data. Machine learning techniques can potentially improve the current methods to identify and isolate measurements from distinct populations. A typical advantage of these techniques is the ability to capture automatically complicated patterns that can be hard to model otherwise; especially in complex and massive datasets as those frequently used in biomedical research. However, introducing these new machine-learning techniques in the field presents several challenges; some of which are validation, adaptation to complex datasets, and identification of the best machine-learning approaches to specific problems.

Here we describe how we addressed these challenges in the context of the L1000 gene-expression assay. L1000 is a new high-throughput gene-expression assay that achieves significant cost reductions compared to more traditional methods, such as RNA-seq, and has contributed to a massive scale-up of the Connectivity Map [@lamb2006connectivity] (a large catalogue of human signatures of genetic or pharmacologic perturbation). Part of L1000's cost reductions stem from a deconvolution procedure, called d-peak, that isolates gene-specific expressions of two genes from a composite measure obtained by a single analyte type [@subramanian2017next]. This procedure works well in practice, but might still be suboptimal for this problem.

Using L1000, we generated a novel experimental dataset with the transcriptional response of approximately 1,000 genes to 122 different perturbagens (shRNA and compounds) and several replicates for a total of over 2,200 gene expression experiments. 
For each experiment, we employed two different detection methods. The first detection method, called UNI, acquired the gene expression of each single gene with one single analyte type (approximately 1,000 $\times$ 2,200 = 2,200,000 distinct measurements). 
The second method, called DUO, acquired a composite measure of the gene expression of two genes by coupling each analyte type to two different genes (approximately 500 $\times$ 2,200 = 1,100,000 distinct measurements). Because the conditions beyond the detection method were nearly identical, the UNI data provided ground truth that enabled us to evaluate empirically different deconvolution methods on the DUO data.

Leveraging this data set, we then explored different machine learning approaches through an open innovation competition, or contest, which we designed following insights from our previous work [@lakhani2013prize; @blasco2019advancing]. We run the contest on Topcoder, a popular crowdsourcing platform. The contest challenged participants to use the novel dataset to improve the deconvolution algorithm utilized by the L1000 platform. The contest drew about 300 competitors from across the globe and resulted in a diverse number of machine-learning approaches. The top approaches included machine-learning methods, such as Random Forests and Convolutional Neural Networks (CNNs), as well as more traditional Gaussian-mixture models. These approaches significantly performed better than the L1000 benchmark in various measures of accuracy and computational speed, and likely have application beyond gene expression.

As we discuss in more detail below, results showed that a Random Forest approach, a popular machine-learning technique, achieved (i) the highest "global" correlation between UNI and DUO data, (ii) the lowest inter-replicate variability and (iii), compared to the benchmark, was able to detect more than a thousand additional differentially-expressed genes, while improving the detection precision at the same time. This provides evidence of the tremendous potential of using machine-learning approaches for deconvolution methods in biology. 


<!-- 
TO READ: 

"DeconvSeq: deconvolution of cell mixture distribution in sequencing data" https://academic.oup.com/bioinformatics/advance-article-abstract/doi/10.1093/bioinformatics/btz444/5506629?redirectedFrom=fulltext

"Evaluation of methods to assign cell type labels to cell clusters from single-cell RNA-sequencing data [version 3; peer review: 2 approved, 1 approved with reservations]"
https://f1000research.com/articles/8-296

Systematic comparative analysis of single cell RNA-sequencing methods
https://www.biorxiv.org/content/10.1101/632216v1

The Technology and Biology of Single-Cell RNA Sequencing
https://www.sciencedirect.com/science/article/pii/S1097276515002610

Fractional proliferation: a method to deconvolve cell population dynamics from single-cell data
https://www.nature.com/articles/nmeth.2138


Bulk tissue cell type deconvolution with multi-subject single-cell expression reference
https://www.nature.com/articles/s41467-018-08023-x

A Single-Cell Transcriptomic Map of the Human and Mouse Pancreas Reveals Inter- and Intra-cell Population Structure
https://www.sciencedirect.com/science/article/pii/S2405471216302666

-->