\appendix
\clearpage 

# Supporting information {-}

```{r annot_ds, include = FALSE}
library(data.table)
annot <- fread("data/holdout_sample_annotations.txt")
gene_annot <- fread("data/barcode_to_gene_map.txt", colClasses = c("gene_id"="character"))
```

\clearpage 

## S1 Figure  {-}

**eCDF of genewise Spearman's rank correlation.** This figure shows the empirical Cumulative Distribution Function (eCDF) of the genewise Spearman's rank correlation for the top-four performing approaches and the k-means benchmark. The top panels present the eCDFs corresponding to the shRNAs, whereas the bottom panels present the eCDFs corresponding to the compounds.

\begin{figure}
\centering
\includegraphics{outcomes/figures/ecdf.corr-shrna.pdf}
\includegraphics{outcomes/figures/ecdf.corr-compounds.pdf}
\end{figure}

\clearpage 

## S2 Figure {-}

**Improvements in genewise Spearman's rank correlation.** This figure shows the association between the genewise Spearman's rank correlation for the top-four performing approaches and the k-means benchmark; it also reports the fraction genes with a higher genewise Spearman's rank correlation than the benchmark. The top panels present the scatter plots corresponding to the shRNAs, whereas the bottom panels present the scatter plots corresponding to the compounds.

\begin{figure}
\centering
\includegraphics{outcomes/figures/corr-shrna.pdf}
\includegraphics{outcomes/figures/corr-compounds.pdf}
\end{figure}

\clearpage 

## S3 Figure {-}

**Improvements in inter-replicate variability.** This figure shows the empirical Cumulative Distribution Function (eCDF) of the genewise inter-replicate variance for the top-four performing approaches and the k-means benchmark. The top panels present the eCDFs corresponding to the genes in low bead proportion (recall genes are paired and mixed in 2:1 ratio within each pair), whereas the bottom panels present the eCDFs corresponding to the genes in high bead proportion.

\begin{figure}
\centering
\includegraphics{outcomes/figures/inter-rep-low.pdf}
\includegraphics{outcomes/figures/inter-rep-high.pdf}
\end{figure}

\clearpage 

## S1 Table. {-}

**Compound perturbagens descriptives.** This table shows componud perturbagen names (`pert_iname`), unique id (`pert_id`), time of treatment (`pert_itime`), dose (`pert_idose`), and number of replicates (`num_replicates`). 

```{r compound_table, echo = FALSE}
annot %>% 
  filter(grepl("BRD", pert_id), handle == "ground-truth") %>%
  count(pert_iname, pert_id, pert_itime, pert_idose) %>%
  rename(num_replicates = n) %>%
  knitr::kable()
```

\clearpage 

## S2 Table. {-}

**Non-compound perturbagens descriptives.** This table shows non-compound perturbagen names (`pert_iname`), unique id (`pert_id`), and number of replicates (`num_replicates`). 

```{r non_compound_table, echo = FALSE}
annot %>% 
  filter(grepl("TRCN", pert_id), handle == "ground-truth") %>%
  count(pert_iname, pert_id) %>%
  rename(num_replicates = n) %>%
  knitr::kable()
```

\clearpage 

## S3 Table {-}

**Top-nine performing solutions.**  This table lists the top-nine solutions and the languages and algorithms each used, as well as the average speedup per plate relative to the k-means benchmark.  

```{r soln_desc, echo=F, warning=F}
library(data.table)
soln_desc <- fread("data/solution_descriptions.txt", na.strings = "n/a")
setnames(soln_desc, "method_short", "category")
knitr::kable(soln_desc[1:9])
```

\clearpage 

## S1 Appendix {-}

**Scoring function.** This appendix describes the scoring function used in the contest to evaluate the performance of the competitors' submissions.

Submissions were scored based on a scoring function that combines measures of accuracy and computational speed. Accuracy measures were obtained by comparing the contestant's predictions, which were derived from $DUO$ data, to the equivalent $UNI$ ground truth data generated from the same samples.

The scoring function combines two measures of accuracy: correlation and AUC, which are applied to deconvoluted ($DECONV$) data and one to differential expression ($DE$) data, respectively.

$DE$ is derived from DECONV by applying a series of transformations (parametric scaling, quantile normalization, and robust z-scoring) that are described in detail in @subramanian2017next. The motivation for scoring $DE$ data in addition to $DECONV$ is because it is at this level where the most bilogically interesting gene expression changes are observed. Of particular interest is obtaining significant improvement in the detection of, so called, "extreme modulations." These are genes that notably up- or down-regulated by pertubation and hence exhibit an exceedingly high (or low) $DE$ values relative to a fixed threshold. 

The first accuracy component is based on the Spearman rank correlation between the predicted $DECONV$ data and the corresponding $UNI$ ground truth data.

For a given dataset $p$, let $M_{\text{DUO},p}$ and $M_{\text{UNI},p}$ denote the matrices of the estimated gene intensities for $G = 976$ genes (rows) and $S = 384$ experiments (columns) under DUO and UNI detection.
Compute the Spearman rank correlation matrix, $\rho$, between the rows of these matrices and take the median of the diagonal elements of the resulting matrix (i.e., the values corresponding to the matched experiments between UNI and DUO) to compute the median correlation per dataset, 
$$
  \text{COR}_p = median(diag(\rho(M_{\text{DUO},p}, M_{\text{UNI},p})). 
$$

The second component of the scoring function is based on the Area Under the receiver operating characteristic Curve (AUC) that uses the competitor's DE values at various thresholds to predict the UNI's DE values being higher than 2 ("high") or lower than -2 ("low").
 
For a given dataset $p$, let $\text{AUC}_{p, c}$ denote the corresponding area under the curve where $c = \{ \text{high}, \text{low} \}$ (either higher than 2 or less than -2); then, compute the arithmetic mean of the area under the curve per class to obtain the corresponding score per dataset:
$$
  \text{AUC}_p = (\text{AUC}_{p,\text{high}} + \text{AUC}_{p,\text{high}} / 2.
$$

These accuracy components were integrated into a single aggregate scores:
$$
\text{SCORE} = 
  \text{SCORE}_{\text{max}} \cdot (\max(\text{COR}_{p}, 0)) 2 
                   \cdot \text{AUC}_{p} 
                   \cdot \exp(- T_{\text{solution}} / (3 \cdot T_{\text{benchmark}})), 
$$ 
where $T_\text{solution}$ is the run time in seconds for deconvoluting the data in each plate, and $T_{\text{benchmark}}$ is the deconvolution time required by the benchmark D-Peak implementation. 

\clearpage

## S2 Appendix  {-}

**L1000 Experimental Scheme** The L1000 assay uses Luminex bead-based fluorescent scanners to detect gene expression changes resulting from treating cultured human cells with chemical or genetic perturbations [Subramanian 2017]. Experiments are performed in 384-well plate format, where each well contains an independent sample. The Luminex scanner is able to distinguish between 500 different bead types, or colors, which CMap uses to measure the expression levels of 978 landmark genes using two detection approaches.

In the first detection mode, called $UNI$, each of the 978 landmark genes is measured individually on one of the 500 Luminex bead colors. In order to capture all 978 genes, two detection plates are used, each measuring 489 landmarks. The two detection platesâ€™ worth of data are then computationally combined to reconstruct the full 978-gene expression profile for each sample.

By contrast, in the $DUO$ detection scheme two genes are measured using the same bead color. Each bead color produces an intensity histogram which characterizes the expression of the two distinct genes. In the ideal case, each histogram consists of two peaks each corresponding to a single gene. The genes are mixed in 2:1 ratio, thus the areas under the peaks have 2:1 ratio (see Figure 1), which enables the association of each peak with the specific gene. The practical advantage of the DUO detection mode is that it uses half of the laboratory reagents as UNI mode, and hence $DUO$ is and has been the main detection mode used by CMap. After $DUO$ detection, the expression values of the two genes are computationally extracted in a process called 'peak deconvolution,' described in the next section. 

