
Introduction
============

\color{red}
TODO: try to rephrase first paragraph
\color{black}

Multianalyte methods for high-throughput gene-expression profiling have found widespread application in biology. The extent of these applications, however, tend to be limited by the type and number of available analytes, which typically results in prohibitive costs for big data generation. Using an assay called L1000 that measures the mRNA transcript abundance of 978 "landmark" genes from human cells, the Connectivity Map (CMap) group at the Broad Institute has developed a novel approach that matches pairs of genes to the same analyte type, or bead, to double the count of profiled genes, thus lowering greatly costs [@subramanian2017next]. 

A central component of this approach is to statistically deconvolve and compare gene type-specific expression profiles for sets of two genes in mixed-gene samples. Similar deconvolution problems are ubiquitous in biology and several solutions have been proposed in a variety of different contexts. Examples include algorithms to identify cell type–specific gene expression differences in complex tissues [@shen2010cell] or dynamic changes in cell populations [@lu2003expression], or ways to discover the target proteins of small molecules [Jung, 2015].

CMap’s current deconvolution algorithm (called dpeak) is based on the k-means clustering algorithm, which automatically partitions a set of gene-expression measurements into $k$ clusters by minimizing the within-cluster sum of squares, and then assigns the two largest clusters to each gene type within each matched pair of genes. 
<!--  of the pair, using additional information about differences in the bead proportions of the genes (within each pair, one gene has a higher proportion of beads than the other). -->

<!-- mean of the largest cluster to the gene in higher proportion and the mean value of the second largest cluster to the gene in lower proportion -->

The current k-means approach works well in practice but has several limitations. It tends to split clusters incorrectly when the distributions are not reasonably well separated, it is sensitive to outliers, and it is computationally demanding (an efficient k-means algorithm, such as Lloyd's, has a running time that scales as $O(kn)$ where $n$ is the number of observations and $k$ the number of clusters). 

As a research tool to test for alternative solutions, we used an open innovation competition [@lakhani2013prize; @blasco2019advancing]. In this paper, we report the outcomes of the competition, which successfully engaged a variety of competitors<!--computer scientist, software developers and bioinformatics--> in the problem, resulting in a cost-effective exploration of competing approaches (random forest, gaussian mixture models, and convolutional nets) that would have been otherwise prohibitive.

<!-- Testing for alternative methods would have required substantial resources to experiment with these alternative approaches (more than what already done) and to adapt new to our data. Moreover, impossible an exhaustive search for all available approaches to try; and the combination of these different approaches.  -->


<!-- 
Shen-orr et al. (2010). Gene expression data from multiple cell-types. The problem is to quantify cell-type frequency "then deconvolve and compare cell type-specific average expression profiles for groups of mixed tissue samples." One approach is "deconvolution to be linear"
 -->

<!-- Broadly speaking, the problem is to estimate the density $f$ of a random variable $X$ based on $n$ observations from $Y = X + \epsilon$, where $\epsilon$ is a measurement error based on a possibly unknown distribution.  (Fan, 1991) -->

<!-- The solution adopted by CMap was a KNN [explain]. This is a standard technique implemented in xxx that gives a good level of accuracy [xxx] and reasonably fast computation time (xxx with the current Matlab implementation). -->
