Introduction
============
 
Many recent examples have shown significant benefits for drug discovery from the systematic analysis of large repositories of gene-expression profiles [Refs]. However, traditional gene-expression high-throughput profiling technologies that are based multianalyte methods, such as Luminex profiling technology, are limited by the type and number of available analytes [Refs]. Therefore, the cost of big data generation in biology remains prohibitive. 

The Connectivity Map (CMap) group at the Broad Institute has developed a novel approach that matches pairs of genes to the same Luminex beads to double the count of profiled genes per bead, thus lowering costs. A central component of this approach is to quantify gene-type frequency of beads, and then statistically deconvolve and compare gene type-specific average expression profiles for pairs of mixed gene samples. 

This type of deconvolution problems are ubiquitous and have a long history in biology [refs]. For example, one common deconvolution is used to identify cell type–specific gene expression differences in complex tissues [Shen-orr et al., 2010]; or the discovery of the target proteins of small molecules [Jung, 2015]. 

<!-- 
Shen-orr et al. (2010). Gene expression data from multiple cell-types. The problem is to quantify cell-type frequency "then deconvolve and compare cell type-specific average expression profiles for groups of mixed tissue samples." One approach is "deconvolution to be linear"
 -->

<!-- Broadly speaking, the problem is to estimate the density $f$ of a random variable $X$ based on $n$ observations from $Y = X + \epsilon$, where $\epsilon$ is a measurement error based on a possibly unknown distribution.  (Fan, 1991) -->

<!-- The solution adopted by CMap was a KNN [explain]. This is a standard technique implemented in xxx that gives a good level of accuracy [xxx] and reasonably fast computation time (xxx with the current Matlab implementation). -->

Common approaches are parametric (mixture models) and non-parametric. CMap's approach was K-nearest neighbors (KNN) approach. [Explain briefly.] This works well but has several problems as well. [List problems.] Including time. The trade-off between accuracy and computation time is currently unknown. 

Alternative methods are well known, such as Gaussian etc. But it would have required substantial resources to experiment with these alternative approaches (more than what already done) and to adapt new to our data. Moreover, impossible an exhaustive search for all available approaches to try; and the combination of these different approaches. 

Instead, we used an open innovation competition as a research tool to engage a variety of computer scientist, software developers and bioinformatics in the problem. This allowed a simultaneous exploration of competing approaches tailored to our problem, at no cost. 


Methods
========

In biomedical research, our focus here, deconvolution problems are common in multianalyte assay methods. These methods are widely used to do X, Y and Z. In general terms, multianalyte assay methods are based on microspheres with different fluorescence decay times. This feature can be used to do X, Y, and Z. [EXPLAIN BRIEFLY CMap PROBLEM].
One problem with existing approaches is that they […. ]

To identify accurate methods we launched an open challenge that allowed a rapid exploration of different approaches. Key ingredients of there challenges are:
training and testing dataset
benchmark solution to improve

## Statistical deconvolution of gene-specific expression profiles.

Assume fluorescent-intensity values $X_{ij}$ for beads $i=1,2,\dots, n$ and analytes $j=1,2,\dots, J$, and gene-specific proportions $w_{ik}$ for beads $i=1,2,\dots, n$ and genes $k=1,2,\dots, K$. Our model of analyte fluourescent intensity is:
\[
  X_{ij} = \sum_{k=1}^{K} w_{ik} h_{kj} + e_{ij}. 
\]
where $h_{ik}$ is the gene-expression value for genes $k=1,2,\dots, K$ and analytes $j=1,2,\dots, J$. 

For the UNI detection method, the gene-specific proportions are such that each analyte has only one gene. Hence, $w^{\text{uni}}_{ik} = 1$ when $j = k$, and it is zero otherwise. This implies that each sample can detect at most $J$ different genes under the UNI method. 

For the DUO detection method, the gene-specific proportions are such that each analyte is paired with two genes in 1:2 ratio. Hence, pick an element $g\in G^2$ from the set $G^2$ of all non-overlapping subsets of size two of the gene set $G$. For each pair of genes in $g$ associated with an analyte $j$, we have: $w^{\text{duo}}_{i1} = 2/3$, $w_{i2}=1/3$ and is zero otherwise. 

<!-- 
Each measurement produces an intensity histogram (a list of integers), which characterizes expression of two distinct genes in the sample (for a total of 488 x 2 = 976 genes). In the ideal case, each histogram consists of two peaks (see Figure 1), each corresponding to a single gene. The genes are mixed in 2:1 ratio, thus the areas under the peaks have 2:1 ratio, which allows us to associate each peak with the specific gene. The median position of each peak corresponds to the gene's expression level, and that's what you need to determine in this challenge.
-->

## Data generation 

To generate data for this contest, we profiled six 384-well perturbagen plates, each containing mutually exclusive sets of compound and shRNA treatments. Multiple treatment types were used to ensure avoid potentially over-fitting to any one. The compound and shRNA perturbagen plates were arbitrarily grouped into pairs, and to avoid any potential ‘information leakage’ each pair was profiled in a different cell line. The resulting lysates were amplified by ligation mediated amplification (LMA, Subramanian 2017). The amplicon was then split and detected in both one-gene-per-bead (UNI) and two-genes-per-bead (DUO) detection modes. The three pairs of data were arbitrarily assigned to training, testing, and holdout categories, where in each case the UNI data served as the ground truth.

The data so generated were then split into three subsets called: training, provisional testing, and system testing. Training and provisional-testing data were made available for all the contestants to develop and validate their solutions, while system-testing data were secured to evaluate competitors' last submissions, which was used to award the prizes, and to avoid overfitting as well. 

```{r, echo=FALSE}
category <- c("training", "provisional testing", "system testing")
type <- c("Compounds", "shRNA")
knitr::kable(expand.grid(Category=category, Type=type), caption="Data generated")
```

## To read

- [Compound signature detection on LINCS L1000 big data](https://pubs.rsc.org/en/content/getauthorversionpdf/c4mb00677a) used a fuzzy c-means Gaussian Mixture Model (GMM) to process raw L1000 data, showing better performance compared to KNN. This method is described below:

> To deconvolute such overlapped peaks, we assumed that the fluorophore intensities of each analyte type (corresponding to a specific mRNA type) had a Gaussian distribution. The distribution of the mixture of analytes GeneH(i) and GeneL(i) corresponding to the expression levels of GeneH and GeneL, respectively, should be subject to a bimodal Gaussian distribution, with the proportion of 1.25 to 0.75. We initialized the estimations of the two Gaussian distributions using buzzy c-means clustering [11] and estimated the GMM parameters using the Nelder-Mead method [12]. Thus, the overlapped peaks were deconvoluted as the two estimated Gaussian peaks and the expression levels of the two genes sharing the same analyte were extracted. Mathematical details are included in the Supplementary Methods (the GMM model).

  
- [Deconvolution of linear systems by constrained regression and its relationship to the Wiener theory](https://ieeexplore.ieee.org/abstract/document/4044778)

- [Efficient Bayesian-based multiview deconvolution](https://www.nature.com/articles/nmeth.2929)

- [A Bayesian deconvolution strategy for immunoprecipitation-based DNA methylome analysis](https://www.nature.com/articles/nbt1414)

- [Gene expression deconvolution in linear space](https://www.nature.com/articles/nmeth.1830)

- [Cell type–specific gene expression differences in complex tissues](https://www.nature.com/articles/nmeth.1439)



