
Introduction
============

<!-- 
  - Traditional methods may be suboptimal, potentially limiting new discoveries
  - ML methods have potential but presents challenges in validation, adaptation, and identification of the best approach
  - For validation e profiled 6 plates worth of data with 122 perturbagens (shRNA and compounds)
-->


<!-- 
A major limitation of multianalytes assays is their limited number of available analyte types. 
In the context of RNA profiling, the CMap group has developed a framework that integrates statistical analysis with traditional multianalyte methods to infer differential expressions for multiple genes from the same analyte type. 
We report the results of an open innovation competition that impvored upon the state-of-the-art by integrating machine-learning techniques to address the deconvultion problem. 
 -->

<!--
Over the last decade, the ability to analyze multiple analytes simultaneously from the same biological sample has contributed to creating large perturbational datasets that are essential to accelerate the discovery of novel therapeutics. However, a fundamental limitation of traditional multianalyte assays lies in the restricted extent and type of available analytes
-->

Biomedical researchers frequently leverage the performance of multiplex assays by using a single analyte, or sensor, to measure multiple signals from distinct populations (cell types, tissues, and genes). Common examples include measuring cell-type specific differential gene expression from mixed samples [@shen2010cell], profiling single-cell gene expression in heterogeneous tissues [@deng2019scalable], and tracking cell growth in dynamically evolving populations [@lu2003expression]; but similar occurrences arise also in many other related contexts, such as imaging [e.g., multiview integration, @preibisch2014efficient]. In all these situations, researchers must deconvolute the composite data to identify and isolate measurements from each population.Â 

Traditional deconvolution methods work well in specific settings but have several limitations as well [ref]. Motivated by the recent success of machine learning in pattern recognition tasks, we hypothesized that these more-recent techniques could be effective in addressing deconvolution problems as well. Compared to more traditional methods, a key advantage of machine learning techniques, such as Neural networks or Random Forests, is that the learning algorithm automatically captures nonlinear patterns that are hard to model otherwise, especially in complex and massive datasets as those frequently used in biomedical research. However, entering these new machine learning techniques in the field presents several challenges. Some of which is validation, adaptation to specific datasets, and identification of the best machine-learning approaches to specific problems.

Here we describe how we addressed these challenges in the context of the deconvolution of gene-specific differential gene expression from samples containing a mixture of multiple genes. Using the L1000 gene-expression assay [@subramanian2017next], we generated a novel experimental dataset with the transcriptional response of approximately 1,000 genes to 122 different perturbagens (shRNA and compounds) with several replicates for a total of over 2,200 gene expression experiments. For each experiment, half of the data were obtained by using one single analyte to measure the differential gene expression of individual genes from a mixed sample of two genes (DUO data); and the other half was obtained by using one analyte per each gene (UNI data). This enabled us to evaluate empirically different deconvolution methods under nearly identical conditions.

Based on these data, we then explored different machine learning approaches through an open innovation competition [@lakhani2013prize; @blasco2019advancing]. We run the contest on a crowdsourcing platform (Topcoder) which is popular in the United States. The contest challenged participants to use the novel dataset to improve the deconvolution algorithm utilized by the L1000 platform. The contest drew competitors from across the globe and resulted in a diversity of machine-learning approaches. The top approaches included machine-learning methods, such as Random Forests and Convolutional Neural Networks (CNNs), as well as more traditional Gaussian-mixture models. These approaches significantly improved over the L1000 benchmark in various measures of accuracy and computational speed, and likely have application beyond gene expression.

Results show that the winning approach based on a random forest, a popular machine-learning technique, achieved (i) the highest "global" correlation between UNI and DUO data, (ii) the lowest inter-replicate variability and (iii), compared to the benchmark, was able to detect more than a thousand additional differentially-expressed genes, while improving the detection precision at the same time. This provides evidence of the tremendous potential of using machine-learning approaches for deconvolution methods in biology. 

<!-- 
Here we describe how we addressed these challenges in the context of the deconvolution of gene-specific expression data using the L1000 gene expression assay [@subramanian2017next]. Using L1000, we generated a novel experimental dataset with the transcriptional response of approximately 1,000 genes to 122 different perturbagens (shRNA and compounds) with several replicates for each. We profiled these data from separate and mixed samples. used for the evaluation of different methods. We then describe the results of an open innovation competition [@lakhani2013prize; @blasco2019advancing] using the generated dataset to improve the deconvolution methods on L1000 data. The contest drew competitors from across the globe and resulted in a diversity of machine-learning approaches that significantly improved over the L1000 benchmark and likely have application beyond gene expression. 
 -->

 


<!-- 
Traditional deconvolution methods have a long history in biology, as well as in many other areas of science [ref]. 
While traditional methods seem to work well in specific applications in biology, they have several limitations as well [ref]. The recent success of machine learning in many pattern recognition tasks suggests that these methods could be effective in addressing deconvolution problems as well. This is because of the ability of ML techniques to discover patterns in complex and massive datasets of the size of many current datasets in biomedical research, such as the Connectivity Map (CMap). However, entering these new machine learning techniques in the field presents several challenges, including validation, adaptation to specific datasets, and identification of the best ML approaches to the problem.
 -->
  
<!-- 
We address theis problem in the context of xxxx. 
    - For evaluation, we generated a unique dataset of gene-expression data with ground truth 
      (which is now publicly available)  
    - To explore different approaches we used an open contest ($20,000 prize pool)    
-->
 
