\clearpage 

Introduction
============
 
Traditional multianalyte methods for gene-expression profiling are usually limited by the type and number of available analytes, which typically results in prohibitive costs for big data generation in biology. Using an assay called L1000, the Connectivity Map (CMap) group at the Broad Institute has developed a novel approach that matches pairs of genes to the same analyte type, or bead, to double the count of profiled genes; thus lowering costs [@subramanian2017next]. 

A central component of this approach is to statistically deconvolve and compare gene type-specific expression profiles for pairs of mixed-gene samples. Similar deconvolution problems are ubiquitous in biology and several solutions have been proposed in a variety of different contexts. Examples include algorithms to identify cell type–specific gene expression differences in complex tissues [@shen2010cell] or dynamic changes in cell populations [@lu2003expression], or ways to discover the target proteins of small molecules [Jung, 2015].

CMap’s current deconvolution algorithm (called "D-Peak") is based on the K-means clustering, which automatically partitions a set of gene-expression measurements into $k$ clusters by minimizing the within-cluster sum of squares, and then assigns the two largest clusters to each gene within each matched pair of genes. 
<!--  of the pair, using additional information about differences in the bead proportions of the genes (within each pair, one gene has a higher proportion of beads than the other). -->

<!-- mean of the largest cluster to the gene in higher proportion and the mean value of the second largest cluster to the gene in lower proportion -->

The current k-means approach works well in practice, but has several limitations as well; it may split clusters incorrectly even when data are reasonably well separated; and it is computationally demanding (finding k clusters takes). So, we used an open innovation competition [@lakhani2013prize; @blasco2019advancing] as a research tool to test for alternative solutions and to engage a variety of computer scientist, software developers and bioinformatics in the problem. Here, we report the results of this cost-effective exploration of competing approaches --- e.g., random forest, gaussian mixture models, and convolutional nets --- tailored to our problem, that would have been otherwise prohibitive. 

<!-- Testing for alternative methods would have required substantial resources to experiment with these alternative approaches (more than what already done) and to adapt new to our data. Moreover, impossible an exhaustive search for all available approaches to try; and the combination of these different approaches.  -->


<!-- 
Shen-orr et al. (2010). Gene expression data from multiple cell-types. The problem is to quantify cell-type frequency "then deconvolve and compare cell type-specific average expression profiles for groups of mixed tissue samples." One approach is "deconvolution to be linear"
 -->

<!-- Broadly speaking, the problem is to estimate the density $f$ of a random variable $X$ based on $n$ observations from $Y = X + \epsilon$, where $\epsilon$ is a measurement error based on a possibly unknown distribution.  (Fan, 1991) -->

<!-- The solution adopted by CMap was a KNN [explain]. This is a standard technique implemented in xxx that gives a good level of accuracy [xxx] and reasonably fast computation time (xxx with the current Matlab implementation). -->
