---
title: "Improving Deconvolution Methods in Biology \
        through Open Innovation Competitions: \
        an Application to the Connectivity Map"

author: 
    - "[INCOMPLETE, ORDER TBD]"
    - Andrea Blasco (Harvard)
    - Ted Natoli (Broad)
    - Rinat A. Sergeev (Harvard)
    - Jin H. Paik (Harvard)
    - Max Macaluso (Broad)
    - Rajiv Narayan (Broad)
    - [...]
    - Karim R. Lakhani (Harvard)
    - Aravind Subramanian (Broad)

abstract: "Deconvolution methods are ubiquitous and have a long history in many areas of science and engineering.   A common problem of deconvolution algorithms is the validation of results using realistic datasets for testing, and the costs involved in tailoring different approaches to such datasets. We describe the deconvolution problem of an assay, L1000, that measures mRNA transcript abundance of approximately 1000 genes from human cells using 500 Luminex beads, such that two transcripts are identified by a single bead color. Using L1000, we generated a novel dataset of transcripts for various perturbagens (shRNA and compounds) where genes were measured both in tandem and individually. We then ran an open competition to elicit different deconvolution algorithms. The competition produced a wide variety of techniques including machine-learning algorithms, such as Convolutional Neural Networks and Random Forests, and more traditional approaches, such as Gaussian mixtures and k-means. All top solutions achieved notable improvements over the benchmark." 

date: 'Last updated: `r format(Sys.Date(), "%b %d, %Y")`'

keywords:
  - biology
  - open innoation competitions
  - crowdsourcing
  - deconvolution
  - gene expressions
  - cell lines

titlepage: false 
linestretch: 1.25
fontfamily: mathpazo
# fontsize: 11pt
# colorlinks: true
# urlcolor: cyan
# citecolor: blue 
geometry: margin=1.25in
# endfloat: true
bibliography: dpeak.bib
biblatexoptions: "sorting=none"
header-includes:
  - \usepackage{bbm}
  - \newcommand\hl{}
  - \newcommand\Fig[1]{{\it Fig. \ref{fig:#1}}}
  - \usepackage{setspace}
  - \onehalfspacing 
---


```{r sections, include = FALSE}
# "instructions.rmd" 
childs <- c("intro.rmd", "methods.rmd"
          , "results.rmd", "discussion.rmd", "figures.rmd", "supporting.rmd")
```

```{r child=childs}
``` 

References
==========

